Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -957,16 +959,31 @@ class BrowserViewController: UIViewController {
         }
     }
     // Mark: Opening New Tabs
+    
+    var isInPrivateMode: Bool {
+        return tabTrayController?.privateMode ?? tabManager.selectedTab?.isPrivate ?? false
+    }
 
     @available(iOS 9, *)
-    func switchToPrivacyMode(isPrivate isPrivate: Bool ){
-        applyTheme(isPrivate ? Theme.PrivateMode : Theme.NormalMode)
-
-        let tabTrayController = self.tabTrayController ?? TabTrayController(tabManager: tabManager, profile: profile, tabTrayDelegate: self)
-        if tabTrayController.privateMode != isPrivate {
-            tabTrayController.changePrivacyMode(isPrivate)
+    func switchToPrivacyMode(isPrivate isPrivate: Bool) -> Success {
+        
+        let success = Success()
+        let tabTrayController = self.tabTrayController ?? TabTrayController(tabManager: self.tabManager, profile: self.profile, tabTrayDelegate: self)
```

Comment:
This feel strange to me - instantiating a new copy of the `TabTrayViewController` just to allow for private mode for authentication. We should reconsider how we performing authentication without the needs of a view controller.